src/multitenant_sdk.js

```diff
@@ -45,13 +47,73 @@ const apis = {
   }),
 };
 
-const authInterceptors = [
+const authTokenInterceptor = {
+  enter: ctx => {
+    const { clientSecret } = ctx;
+
+    return {
+      ...ctx,
+      authToken: {
+        access_token: clientSecret,
+        token_type: 'bearer'
+      }
+    }
+  },
+};
+
+const tokenInterceptors = (authApiEndpointInterceptors) => [
   // TODO: encode secret as JWT
+  new FormatHttpResponse(),
   new AddClientSecretToParams(),
   new SaveToken(),
   new AddAuthTokenResponse(),
+  ..._.get(authApiEndpointInterceptors, 'token')
+];
+
+const clientDataInterceptors = (authApiEndpointInterceptors) => [
+  new FormatHttpResponse(),
+  authTokenInterceptor,
+  new AddAuthHeader(),
+  ..._.get(authApiEndpointInterceptors, 'clientData')
 ];
 
+const tokenAndClientDataInterceptor = (authApiEndpointInterceptors) => (
+  {
+    enter: ctx => {
+      const { tokenStore } = ctx;
+      return Promise.resolve()
+          .then(tokenStore.getToken)
+          .then(storedToken => {
+            // If there's a token with any access, it's only necessary
+            // to fetch the client data. Else, we request a token and 
+            // the response will also contain the client data.
+            // We don't need to distinguish between token scopes.
+            if (storedToken) {
+              return contextRunner(clientDataInterceptors(authApiEndpointInterceptors))(ctx)
+                .then(newCtx => {
+                  const { res } = newCtx;
+                  return {
+                    ...newCtx,
+                    res: {
+                      ...res,
+                      data: {
+                        access_token: storedToken.access_token,
+                        client_data: res.data
```

Comment:
I think we need to do something more here or in the `AddAuthTokenResponse` interceptor (or a new one in its place). Currently, the `AddAuthTokenResponse` takes all data from the token endpoint and places it in the token store, as it assumes all data in the response is about the token. That's `access_token`, `refresh_token`, etc. SInce our endpoint returns also client data, we need to exclude that and prevent that data from leaking to the token store. Perhaps, best is to explicitly enumerate the stored data:

* access_token
* scope
* token_type
* expires_in
* the optional refresh_token won't be there in multitenant responses actually

Alternatively, we could change the shape of the response of the API a bit and have two global keps: `token` and `client_data` to better separate which is part of the token response and what is client config. What do you think?

Reply:
Yes, we need to verify that. Should we do it in this PR, or in a separate that just deals with the token response handling?

Reply:
Let's do it here, I think, so that we don't miss it and leak secrets later on :)

Reply:
Just thought that it's a bit unrelated to the client data fetching, and to avoid dragging this too much, since other changes will depend on this. But ok, I'll do it here.

Reply:
Unrelated, but still produces unwanted effect :) Sure, we'd not use the intermediate implementation, but somehow feels wrong to leave in :) Thanks for addressing!